Prosthesis Changes & Mechanical Surgery Tweaks

Skies-Of-Blue opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Kept you waiting, huh?

Firstly, the smaller changes:
- Prosthesis organ manipulation now uses a crowbar instead of a hemostat to remove organs. This was one of two instances of surgical tools being required for IPC Surgery, and it always felt out of place.
- Adds a new surgery step, "Close hatch," has been added to most mechanical surgeries for the sake of consistency. This affects implant removal, ipc revival, mechanical brain surgery, and prosthesis organ manipulation. The new steps can be found in the changelog. 
- Removes player-facing access to chainsaw arms and synthetic arm blades. They always felt ill-fitting for our setting. The code remains, though, just in case someone wants to add untraditional limbs in the future.

And now, for the main course: 

This PR creates two new surgeries, Detach Prosthesis and Prosthesis Attachment, which allow any character with a prosthesis to remove and attach it themself with a set of basic tools.

Additionally, it renames Prosthetic Replacement surgery to Limb Grafting surgery, to better reflect its new role as the more surgical variant of limb attachments. Limb Grafting retains all of its previous functionality, minus chainsaw attachments.

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
Hoo boy, this PR was a long time coming. 

It's always been silly to me that characters are forced to keep their prostheses on 24/7. I can't imagine how uncomfortable that must get. Being forced to have a surgeon remove your leg (or to chop it off with something sharp!!!) just to give their residual limb a breather has always been stifling for prosthesis-based roleplay. This should hopefully open up more options for player expression, and help distinguish standard limbs from mechanical ones.

The two new procedures added serve as alternatives for the more surgical Amputation/Grafting duo we've all grown to know and resent, rather than an outright replacement. It's 505 FSC, and prostheses can range from Glorified Peg Leg all the way to Fully Mobile Prosthetic Hand With Synthetic Nerves. As such, I think it's important to keep a couple variants of surgery around for that mechanical expression, even if they end with the same result. This also allows the Friendly Neighborhood Li-Teg to charge extra for limb grafting surgery as opposed to a basic prosthesis attachment procedure :) 

It's always bugged me that IPCs had to use a hemostat to remove their organs, and I've heard that annoyed sentiment echoed in others over the years. Swapping it gives more consistency to their mechanics overall, and allows for more IPCs to horrifically disassemble themselves as a form of self-soothing while the captain watches in awe.

Lastly, chainsaw arms were a meme and have always felt too /TG/ for our setting. Were Shiptest a codebase built from the ground up, I doubt we would have ever seen these added to the game.

I know some players expressed interest in having surplus limbs be attach/removable without tools of any sort, but due to time constraints and a general lack of my own abilities, that remains out of reach for the time being. Maybe I'll look into it in the future, who knows.
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog

:cl:
add: creates close_hatch, a step that mirrors open_hatch in prosthetic surgeries for the sake of consistency. Why you leaving that hatch open?? The final step before screwing to finish surgery. This affects implant removal, ipc revival, mechanical brain surgery, and prosthesis organ manipulation
add: creates manipulate_organs/mechanic, a step that swaps the necessary implements in synthetic surgery. Use a crowbar instead of a hemostat to remove any organs on an augmented limb!
tweak: changes implant removal surgery to reflect the above. The new tool order is as follows: screwdriver, hand, wrench, crowbar, wrench, hand, screwdriver
tweak: changes the (unused) ipc revival surgery to reflect the above. The new tool order is as follows: screwdriver, hand, multitool, inducer, hand, screwdriver
tweak: changes mechanical brain surgery to reflect the above. The new tool order is as follows: screwdriver, hand, multitool, hemostat, hand, screwdriver
tweak: changes prosthesis organ manipulation to reflect the above. The new tool order is as follows: screwdriver, hand, wrench (if head/chest), multitool, crowbar, wrench (if head/chest), hand, screwdriver
add: creates the Detach Prosthesis surgery, a less violent alternative to Amputation. This allows for prosthesis users to remove their own limbs, for the purposes of maintenance or just plain comfort! The tool order is as follows: screwdriver, hand, multitool, wrench, crowbar/wrench/hand (so you can flavor how you remove the limb)
tweak: changes the Prosthetic Replacement surgery, renaming it to Limb Grafting surgery. The steps remain the same, but the name needed some retooling, as the procedure is much more surgically involved than its new sibling:
add: creates Prosthesis Attachment surgery, a less surgically involved version of the Limb Grafting surgery. This allows for people with prostheses to re-attach the limb they took off with Prosthesis Removal! The tool order is as follows: wrench, multitool, plug in prosthesis, hand, screwdriver
del: removes the chainsaw and synthetic arm blade from being viable prostheses. Left the code intact in case someone wants to do something similar in-future
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Skies-Of-Blue committed:
come back after towels

Skies-Of-Blue committed:
haha whoops

JimKil3 commented on code/modules/surgery/mechanic_steps.dm:
```diff
@@ -112,3 +112,93 @@
+				"<span class='notice'>[user] begins to replace [target]'s [parse_zone(target_zone)].</span>")
+		else
+			to_chat(user, "<span class='warning'>[tool] isn't the right type for [parse_zone(target_zone)].</span>")
+			return -1
```
this could be made into a guard clause

JimKil3 commented on code/modules/surgery/mechanic_steps.dm:
```diff
@@ -112,3 +112,93 @@
+
+/datum/surgery_step/add_prosthetic/success(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery, default_display_results = FALSE)
+	. = ..()
+	if(istype(tool, /obj/item/bodypart) && user.temporarilyRemoveItemFromInventory(tool))
```
er, isn't the type of the tool already checked by this point? and you could check the `NODROP` flag instead of using that proc

JimKil3 replied:
oops, `NODROP` isn't a flag, it's a trait (`TRAIT_NODROP`) - point still stands

JimKil3 commented on code/modules/surgery/mechanic_steps.dm:
```diff
@@ -112,3 +112,93 @@
+			return -1
+
+/datum/surgery_step/add_prosthetic/success(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery, default_display_results = FALSE)
+	. = ..()
```
```suggestion
```
you dont want to call parent here because there's still a possibility of the step failing during this proc and the parent gives xp

JimKil3 commented on code/modules/surgery/mechanic_steps.dm:
```diff
@@ -112,3 +112,93 @@
+	if(istype(tool, /obj/item/bodypart) && user.temporarilyRemoveItemFromInventory(tool))
+		var/obj/item/bodypart/L = tool
+		if(!L.attach_limb(target))
+			display_results(user, target, "<span class='warning'>You fail in replacing [target]'s [parse_zone(target_zone)]! Their body has rejected [L]!</span>",
```
```suggestion
			display_results(user, target, "<span class='warning'>You fail to replace [target]'s [parse_zone(target_zone)]! Their body has rejected [L]!</span>",
```

JimKil3 commented on code/modules/surgery/mechanic_steps.dm:
```diff
@@ -112,3 +112,93 @@
+		if(!L.attach_limb(target))
+			display_results(user, target, "<span class='warning'>You fail in replacing [target]'s [parse_zone(target_zone)]! Their body has rejected [L]!</span>",
+				"<span class='warning'>[user] fails to replace [target]'s [parse_zone(target_zone)]!</span>",
+				"<span class='warning'>[user] fails to replaces [target]'s [parse_zone(target_zone)]!</span>")
```
```suggestion
				"<span class='warning'>[user] fails to replace [target]'s [parse_zone(target_zone)]!</span>")
```

JimKil3 commented on code/modules/surgery/mechanic_steps.dm:
```diff
@@ -112,3 +112,93 @@
+				"<span class='warning'>[user] fails to replace [target]'s [parse_zone(target_zone)]!</span>",
+				"<span class='warning'>[user] fails to replaces [target]'s [parse_zone(target_zone)]!</span>")
+			L.forceMove(target.loc)
+			return
```
```suggestion
			return FALSE
```

JimKil3 commented on code/modules/surgery/mechanic_steps.dm:
```diff
@@ -112,3 +112,93 @@
+		display_results(user, target, "<span class='notice'>You succeed in replacing [target]'s [parse_zone(target_zone)].</span>",
+			"<span class='notice'>[user] successfully replaces [target]'s [parse_zone(target_zone)] with [tool]!</span>",
+			"<span class='notice'>[user] successfully replaces [target]'s [parse_zone(target_zone)]!</span>")
+		return
```
```suggestion
		return ..()
```

JimKil3 commented on code/modules/surgery/mechanic_steps.dm:
```diff
@@ -112,3 +112,93 @@
+			"<span class='notice'>[user] successfully replaces [target]'s [parse_zone(target_zone)] with [tool]!</span>",
+			"<span class='notice'>[user] successfully replaces [target]'s [parse_zone(target_zone)]!</span>")
+		return
+	return ..() //if for some reason we fail everything we'll print out some text okay?
```
```suggestion
```
you should make the istype and tempremoveitem etc proc into a guard clause

JimKil3 commented on code/modules/surgery/mechanical.dm:
```diff
@@ -99,3 +100,27 @@
+
+/datum/surgery/prosthesis_attachment/can_start(mob/user, mob/living/carbon/target)
+	if(!iscarbon(target))
+		return 0
```
```suggestion
		return FALSE
```

JimKil3 commented on code/modules/surgery/mechanical.dm:
```diff
@@ -99,3 +100,27 @@
+		return 0
+	var/mob/living/carbon/C = target
+	if(!C.get_bodypart(user.zone_selected)) //can only start if limb is missing
+		return 1
```
```suggestion
		return TRUE
```

Skies-Of-Blue committed:
requested changes

Skies-Of-Blue committed:
HAHA WHOOPS FUCK ME SIDEWAYS
Removes the ability to remove robotic heads, allowing FBPs to kill themselves instantly.

Skies-Of-Blue committed:
a note to those who follow

Skies-Of-Blue commented:
If no other issues are reported this uptime, this PR should be more or less ready to merge. I know https://github.com/shiptest-ss13/Shiptest/pull/2811 is waiting on this to go through for their own work.
